zokrates_core/src/semantics.rs

```diff
@@ -3952,18 +3970,18 @@ mod tests {
 
         #[test]
         fn duplicate_function_declaration_generic() {
-            // def foo<P>(private field[P] a) {
+            // def foo<P>(field[P] a) {
             //   return;
             // }
-            // def foo(private field[3] a) {
+            // def foo(field[3] a) {
             //   return;
             // }
             //
             // should succeed as P could be different from 3
 
             let mut f0 = function0();
 
-            f0.value.arguments = vec![untyped::Parameter::private(
+            f0.value.arguments = vec![untyped::Parameter::public(
```

Comment:
Ok this seems to be related to what I mentioned above, the visibility should be an option on each parameter in the untyped ast, and in the typed ast it should probably only appear in TypedProgram, not on each function if possible.